apps/prairielearn/src/components/InstructorInfoPanel.html.ts

```diff
@@ -0,0 +1,372 @@
+  IntervalSchema,
+  type Assessment,
+  type AssessmentInstance,
+  type Course,
+  type CourseInstance,
+  type Group,
+  type InstanceQuestion,
+  type Question,
+  type Variant,
+} from '../lib/db-types.js';
+
+import { QuestionContext } from './QuestionContainer.types.js';
+
+export function InstructorInfoPanel({
+  course,
+  course_instance,
+  assessment,
+  assessment_instance,
+  instance_question,
+  question,
+  variant,
+  user,
+  instance_group,
+  instance_group_uid_list,
+  instance_user,
+  authz_data,
+  question_is_shared,
+  questionContext,
+  csrfToken,
+}: {
+  course: Course;
+  course_instance?: CourseInstance;
+  assessment?: Assessment;
+  assessment_instance?: AssessmentInstance & { formatted_date: string };
+  instance_question?: InstanceQuestion & {
+    assigned_grader_name?: string | null;
+    last_grader_name?: string | null;
+  };
+  question?: Question;
+  variant?: Variant & { formatted_date: string };
+  user: User;
+  instance_group?: Group | null;
+  instance_group_uid_list?: string[] | null;
+  instance_user?: User | null;
+  authz_data: Record<string, any>;
+  question_is_shared?: boolean;
+  questionContext: QuestionContext;
+  csrfToken: string;
+}) {
+  // Here, we are only checking if the effective user is an instructor. We are not
+  // attempting to check if this user has permission to view student data. That check
+  // would have already been made (when necessary) before granting access to the page
+  // on which this partial is included (an assessment instance or question instance).
+  if (
+    !authz_data.has_course_permission_preview &&
+    !authz_data.has_course_instance_permission_view
+  ) {
+    return '';
+  }
+
+  return html`
+    <div class="card mb-4 border-warning">
+      <div class="card-header bg-warning">Staff information</div>
+      <div class="card-body">
+        ${StaffUserInfo({ user })}
+        ${InstanceUserInfo({ instance_user, instance_group, instance_group_uid_list })}
+        ${QuestionInfo({ course, course_instance, question, variant, question_is_shared })}
+        ${VariantInfo({ variant })} ${IssueReportButton({ variant, csrfToken, questionContext })}
+        ${AssessmentInstanceInfo({ assessment, assessment_instance })}
+        ${ManualGradingInfo({ instance_question, assessment, questionContext })}
+      </div>
+      <div class="card-footer small">This box is not visible to students.</div>
+    </div>
+  `;
+}
+
+function StaffUserInfo({ user }: { user: User }) {
+  return html`
+    <h5 class="card-title">Staff user:</h5>
+    <div class="d-flex flex-wrap pb-2">
+      <div class="pr-1">${user.name}</div>
+      <div class="pr-1">${user.uid}</div>
+    </div>
+  `;
+}
+
+function InstanceUserInfo({
+  instance_user,
+  instance_group,
+  instance_group_uid_list,
+}: {
+  instance_user?: User | null;
+  instance_group?: Group | null;
+  instance_group_uid_list?: string[] | null;
+}) {
+  if (instance_user == null && instance_group == null) return '';
```

Comment:
This is slightly different functionality. Previously, this block was only shown if an assessment instance was involved (i.e., for student view and manual grading), and the assessment instance did not belong to the current user. The assessment instance restriction remains (it's the only one that retrieves the values of instance_user/instance_group), but this is now shown for all assessment instances, regardless of if they belong to the current user.